polkadot/runtime/common/src/claims.rs

```diff
@@ -171,7 +176,14 @@ pub mod pallet {
 	pub trait Config: frame_system::Config {
 		/// The overarching event type.
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
-		type VestingSchedule: VestingSchedule<Self::AccountId, Moment = BlockNumberFor<Self>>;
+		type VestingSchedule: VestingSchedule<
+			Self::AccountId,
+			Moment = BlockNumberFor<Self>,
+			Fungible = Self::Fungible,
+		>;
+		type Fungible: fungible::Inspect<Self::AccountId, Balance = Self::Balance>
+			+ fungible::Mutate<Self::AccountId>;
+		type Balance: Balance + MaybeSerializeDeserialize;
```

Comment:
No no, `Balance` can be derived from `type Fungible`.

Reply:
Agreed but the pallet needs `Balance` to be `MaybeSerializeDesirialize`. There alas isn't another way to enforce this bound without pulling `Balance` out into a separate associated type that I know of.

Reply:
Ah yes, the trait bound on the pallet seems to work nicely. Perfect.